serving/ingress/pkg/reconciler/ingress/resources/route.go

```diff
@@ -92,7 +95,9 @@ func makeRoute(ci *networkingv1alpha1.Ingress, host string, rule networkingv1alp
 	}
 
 	labels := kmeta.UnionMaps(ci.Labels, map[string]string{
-		networking.IngressLabelKey: ci.GetName(),
+		networking.IngressLabelKey:      ci.GetName(),
+		OpenShiftRouteLabelKey:          ci.GetName(),
+		OpenShiftRouteNamespaceLabelKey: ci.GetNamespace(),
```

Comment:
Why are we adding these? :thinking:

Reply:
Ingress generated by DomainMapping does not have any label so `serving.knative.dev/{route,routeNamespace}` labels are not added to OCP Route.
Due to this, route lister here https://github.com/openshift-knative/serverless-operator/blob/master/serving/ingress/pkg/reconciler/ingress/ingress.go#L121-L128 cannot list OCP Route for DomainMapping :disappointed_relieved: 

I had a few other ideas:
#### option-a. Use fixed namespace for the lister like:
```
        return r.routeLister.Routes("knative-serving-ingress").List(
                selector,    // networking.IngressLabelKey=ing.GetName()
        )
```
#### b. option-b. Add label to Ingress created by DomainMapping

https://github.com/knative/serving/pull/10370 (But we need to add `serving.knative.dev/domainmappingNamespace` which is not necessary in upstream.)

But I wonder that adding the new label is simple.

Reply:
Most importantly, we can't easily change labels as that'd break potential upgrade cases as the lister can't find the correct routes anymore.

I feel like if anything, we should invest in our component to be less dependent on upstream labels :thinking:. Should we just add our own `ingressName` and `ingressNamespace` labels and don't care about upstream's labels at all?

Reply:
I guess that's kindof what you're doing with the new labels you add. I'd just rename them to be

```
OpenShiftIngressLabelKey          = "serving.knative.openshift.io/ingressName"
OpenShiftIngressNamespaceLabelKey = "serving.knative.openshift.io/ingressNamespace"
```

Reply:
Thank you! As discussed on Slack we write bridging code to adjust routeList to do 2 list operations:
1. the one it already has
2. the new one
3. combine the two